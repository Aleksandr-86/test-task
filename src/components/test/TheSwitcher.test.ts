import { render } from '@testing-library/vue'
import TheSwitcher from '@/components/TheSwitcher.vue'

describe('Компонент TheSwitcher', () => {
  it('структура соответствует снимку', () => {
    const wrapper = render(TheSwitcher, {})

    expect(wrapper).toMatchSnapshot()
  })
})
