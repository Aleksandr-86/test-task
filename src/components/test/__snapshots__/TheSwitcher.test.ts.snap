// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Компонент TheSwitcher > структура соответствует снимку 1`] = `
{
  "baseElement": <body>
    <div>
      <label
        class="relative flex h-[28px] w-max cursor-pointer select-none items-center"
      >
        <input
          class="h-7 w-14 cursor-pointer appearance-none rounded-full border-2 border-gray-400 transition-colors focus:outline-blue-700 dark:border-transparent dark:checked:bg-blue-700 dark:checked:outline-blue-700 dark:focus:outline-double dark:focus:outline-blue-700"
          name="switcher"
          type="checkbox"
        />
        <span
          class="absolute right-7 h-7 w-7 transform rounded-full bg-gray-400 dark:bg-gray-300 transition-transform"
        />
      </label>
    </div>
  </body>,
  "container": <div>
    <label
      class="relative flex h-[28px] w-max cursor-pointer select-none items-center"
    >
      <input
        class="h-7 w-14 cursor-pointer appearance-none rounded-full border-2 border-gray-400 transition-colors focus:outline-blue-700 dark:border-transparent dark:checked:bg-blue-700 dark:checked:outline-blue-700 dark:focus:outline-double dark:focus:outline-blue-700"
        name="switcher"
        type="checkbox"
      />
      <span
        class="absolute right-7 h-7 w-7 transform rounded-full bg-gray-400 dark:bg-gray-300 transition-transform"
      />
    </label>
  </div>,
  "debug": [Function],
  "emitted": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "html": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
